src/main/anovos/data_analyzer/validations.py

```diff
@@ -0,0 +1,187 @@
+from functools import wraps
+from inspect import getcallargs
+from pyspark.sql import functions as F
+from pyspark.sql import types as T
+from anovos.shared.utils import attributeType_segregation, discrete_attributes
+
+
+def refactor_arguments(func):
+    @wraps(func)
+    def wrapper(*args, **kwargs):
+        all_kwargs = getcallargs(func, *args, **kwargs)
+        idf = all_kwargs.get("idf")
+        list_of_cols = all_kwargs.get("list_of_cols")
+        drop_cols = all_kwargs.get("drop_cols")
+
+        num_cols, cat_cols, other_cols = attributeType_segregation(idf)
+        if func.__name__ in (
+            "missingCount_computation",
+            "measures_of_counts",
+            "measures_of_centralTendency",
+            "duplicate_detection",
+            "nullRows_detection",
+            "nullColumns_detection",
+            "correlation_matrix",
+            "variable_clustering",
+            "IV_calculation",
+            "IG_calculation",
+        ):
+            all_valid_cols = num_cols + cat_cols
+        elif func.__name__ in (
+            "measures_of_dispersion",
+            "measures_of_counts",
+            "nonzeroCount_computation",
+            "measures_of_percentiles",
+            "measures_of_shape",
+            "outlier_detection",
+        ):
+            all_valid_cols = num_cols
+        elif func.__name__ in (
+            "uniqueCount_computation",
+            "mode_computation",
+            "measures_of_cardinality",
+            "IDness_detection",
+            "biasedness_detection",
+            "invalidEntries_detection",
+        ):
+            all_valid_cols = discrete_attributes(idf)
+        else:
+            all_valid_cols = num_cols + cat_cols + other_cols
```

Comment:
I don't think that this approach is a good idea for three reasons:

1) It relies on `__name__`, which is a Python internal.
2) It splits up the way the actual function is defined and hides an important part from the user looking at the code. If someone renames the function or changes its arguments, it breaks this decorator.
3) This decorator is very complex, as it tries to manage the input of a large number of functions and their different arguments in just one large function.

Instead of using a decorator, I suggest to follow the principles "Explicit is better than implicit." and "Simple is better than complex." proposed in [The Zen of Python](https://peps.python.org/pep-0020/) and create a set of small, testable, easy-to-read functions that each deal with one particular argument.

For example:

```python
def parse_columns(list_of_cols: Union[str, List[str]]):
    ...

def parse_boolean(boolean_arg: Union[str, bool, int]):
   ...
```

Each of these functions should be independently tested.

Then, use the helper functions within each of Anovos' functions:

```python
def feature_engineering_method(list_of_cols, treatment, ...):

    list_of_cols = parse_columns(list_of_cols)
    treatment = parse_boolean(treatment)
   ...
```

This is easy to understand, easy to test/debug and easy to maintain.

Reply:
```python
@numerical_columns(arg_name="list_of_cols")
@...
@...
def feature_engineering_method(list_of_cols):
     ...
```